hangman.py

```diff
@@ -0,0 +1,173 @@
+
+        # checking and creating the directory if necessary
+        pathlib.Path('./word_data/').mkdir(parents=True, exist_ok=True)
+
+        # storing the relative file path to the words.txt file
+        filePath = './word_data/words.txt'
+
+        try:
+            # checking if the file exists
+            with open(filePath) as f:
+                print('file exists')
+        except FileNotFoundError:
+            # downloading the file to the directory if it does not exist
+            print('file does not exist')
+            url = 'http://norvig.com/ngrams/sowpods.txt'
+            urllib.request.urlretrieve(url, filePath)
+            print('file downloaded to ' + filePath)
+        finally:
+            # storing the file contents as a list
+            with open(filePath) as f:
+                self.lines = f.readlines()
+            print('file read')
+
+    def wordPick(self):
+        # This method randomly selects a word of a minimum length from the stored list
+        # self.minlength = int(args.minlength)
+        self.word = str()  # word chosen by game
+
+        # selecting a random word from the list that has a length greater than the specified minimum
+        while len(self.word) < self.minlength:
+            self.word = random.choice(self.lines).lower()
+            self.word = self.word[:-1]
+
+    def printhangman(self, wrongGuesses):
+        # This method prints on of the 7 states of the hangman
+        # This method prints on of the 7 states of the hangman
+        print('  |‾‾‾‾‾‾‾|')  # printing top line of gallows for all states
+
+        if wrongGuesses > 0:  # printing second line of gallows for all states
+            print('  |       O')
+        else:
+            print('  |')
+
+        if wrongGuesses == 6:  # printing third line of gallows for all states
+            print('  |      /|\\')
+        elif wrongGuesses == 5:
+            print('  |      /|')
+        elif wrongGuesses >= 2:
+            print('  |       |')
+        else:
+            print('  |')
+
+        if wrongGuesses >= 4:  # printing fourth line of gallows for all states
+            print('  |      / \\')
+        elif wrongGuesses == 3:
+            print('  |      / ')
+        else:
+            print('  |')
+
+        print('__|____\n')  # printing fifth line of gallows for all states
+
+    def newGame(self):
+
+        self.openFile()
+        self.wordPick()
+
+        letters = list(string.ascii_lowercase)
+        guessed = copy.deepcopy(letters)
+        guesses = ''
+        target = ''
+        output = ''
+        stateHM = 0
+
+        for char in self.word:
+            target += ' ' + char + ' '
+
+        for char in self.word:
+            output += ' _ '
+        print(output)
+
+        while stateHM < 6:
+            output = ''
+
+            print(*guessed, sep=' ')
+            guess = input("Choose a letter: ")
+            guess = guess.lower()
+
+            if guess in letters:
```

Comment:
This if block is pretty confusing. We have guess, guessed, guesses, letters, and word. Might try to add more clarity to variable name. I am having trouble following the logic with the similar names. We may be able to simplify this block with clearer variable names